doc/src/Thumbnails.dox

```diff
@@ -0,0 +1,57 @@
+/*!
+ * @page thumbnails Thumbnails
+ *
+ * @section thumbnails_overview Overview
+ *
+ * OAIO provides a mechanism for a @ref manager to request a thumbnail
+ * from the @ref host when a new @ref entity is created, and for a host
+ * to look up a thumbnail for an existing entity.
+ *
+ * @section thumbnails_creation_during_publish Requesting Thumbnail Creation During Publishing
+ *
+ * In many situations, it is difficult for a @ref manager to
+ * retrospectively generate a thumbnail for any given @ref entity. It
+ * usually requires first-class knowledge of how the specific entity
+ * type can be rendered to into an image.
+ *
+ * For image-based entities, this is potentially straight forward. The
+ * whole purpose of OAIO though is to allow generic handling of new
+ * entity types without specific logic within the manager.
+ *
+ * It seems fair then, to ask the @ref host to provide a thumbnail with
+ * the entity when it is @ref publish "published". More often than not,
+ * the host not only understands how this might be done, but has the
+ * relevant data to hand to do it.
+ *
+ * To this end, a @ref manager can set the @ref openassetio.constants.kWantsThumbnail
+ * "kWantsThumbnail" flag in it's response to @ref openassetio.managerAPI.ManagerInterface.ManagerInterface.managementPolicy
+ * "managementPolicy" for any given entity @ref Specification.
+ *
+ * If supported by the host, it will then:
+ * - Call @ref preflight with the target entity and a @needsref ThumbnailSpecification
+ *   that describes supported formats.
+ * - Call @ref openassetio.managerAPI.ManagerInterface.ManagerInterface.getEntityMetadata
+ *   "getEntityMetadata" on the returned reference to determine the desired width/height/format
+ *   for the thumbnail.
+ * - Generate a thumbnail and @ref register it to the same reference.
+ *
+ * @note The thumbnail generation process may be carried out
+ * asynchronously by the host and registration may occur at an arbitrary
+ * time in the future.
+ *
+ * @see example_generating_a_thumbnail
```

Comment:
[fix] This ref doesn't render.

Reply:
Fixed [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/116/commits/f16b1413aaa134ce202f67352f1fa6e597ebf15a).